enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,437 @@
+authors:
+  - "@pacevedom"
+reviewers:
+  - "@eslutsky"
+  - "@copejon"
+  - "@ggiguash"
+  - "@pmtk"
+  - "@pliurh"
+  - "@jerpeter1"
+  - "@Miciah"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - None
+creation-date: 2024-01-08
+last-updated: 2024-01-24
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-2089
+---
+
+# MicroShift router default configuration options
+
+## Summary
+MicroShift's default router is created as part of the platform, but does not
+allow configuring any of its specific parameters. For example, you can not
+disable the router or change its listening ports.
+
+In order to allow these operations and many more, a set of configuration options
+is proposed.
+
+## Motivation
+MicroShift's default ingress router comes as part of the product and is always
+deployed. As part of the start procedure, MicroShift will configure the router
+with fixed parameters, such as being always enabled, and tied to ports 80 and
+443. None of these are configurable for the user, and they also tie to how the
+router is exposed in the cluster.
+
+In order to allow more flexibility, the following configuration parameters shall
+be added:
+- Enable/Disable. Some use cases may be egress only, meaning they do not need
+  an ingress controller. In this case the router pod should not be scheduled
+  to save resources. There should be no rules associated to firewalld or
+  iptables either for an improved security posture.
+- Listening ports. Allow configuring ports other than 80 and 443 improves
+  flexibility.
+- Which IPs the router listens on. Some use cases may require the router to be
+  reachable only from certain networks.
+
+### User Stories
+As a MicroShift user, I want to enable the default router with a configuration
+option.
+
+As a MicroShift user, I want to disable the default router and have all the
+manifests automatically removed.
+
+As a MicroShift user, I want to be able to configure the ports on which the
+default router is listening.
+
+As a MicroShift user, I want to be able to choose the IPs where the router is
+listening.
+
+As a MicroShift user, I want to be able to allow traffic to the router from
+specific IP addresses.
+
+### Goals
+* Allow users to enable the router.
+* Allow users to disable the router.
+* Allow users to reach the router in the configured ports.
+* Allow users to configure in which IPs the router listens.
+* Allow users to allow traffic from specific IP addresses.
+* Allow users to deny traffic to the router.
+
+### Non-Goals
+* N/A
+
+## Proposal
+Each configuration option will get its own section with specifics and
+details.
+As a common baseline, a new top level section will be added to the
+configuration:
+```yaml
+apiServer:
+   ...
+# new from here.
+ingress:
+  status: <Enabled|Disabled> # Defaults to Enabled.
+  ports:
+    http: <int> # Defaults to 80.
+    https: <int> # Defaults to 443.
+  expose:
+    - <IP address/NIC> # Defaults to the NICs of the node
+```
```

Comment:
I'd consider this an extension to the MicroShift API since it's user facing. I don't think it's worth duplicating in full under the API Extension section, but it's worth mentioning in short.